addon/components/sl-date-time.js

```diff
@@ -105,9 +105,15 @@ export default Ember.Component.extend( TooltipEnabled, {
         'timezoneString',
         'value',
         function() {
-            return window.moment( this.get( 'value' ) )
-                .format( 'YYYY-MM-DD HH:mm ' ) +
-                this.get( 'timezoneString' );
+            const timezoneString = this.get( 'timezoneString' );
+            const value = this.get( 'value' ) ? this.get( 'value' ) : new Date();
+
+            if ( timezoneString ) {
+                return window.moment.tz( value, timezoneString )
+                    .format( 'YYYY-MM-DD HH:mm z' );
+            }
+
+            return value.format( 'YYYY-MM-DD HH:mm Z' );
```

Comment:
Due to `value` (https://github.com/theoshu/sl-ember-components/blob/issue510/addon/components/sl-date-time.js#L90) defaulting to `new Date()` and the comments made about `timezoneString` in this same PR this computed property will always return a value, negating the need for any of these changes.